zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -329,39 +331,45 @@ benchmarks! {
         // c = num. asset types
         let c in (T::MinCategories::get() as u32)..(T::MaxCategories::get() as u32);
         // d = num. disputes
-        let d in 0..T::MaxDisputes::get() as u32;
+        let d in 0..T::Court::max_disputes() as u32;
 
         let c_u16 = c.saturated_into();
         let (caller, marketid) = setup_resolve_common_categorical::<T>(a, b, c_u16)?;
 
         for i in 0..c.min(d) {
-            let _ = Call::<T>::dispute(marketid, OutcomeReport::Categorical(i.saturated_into()))
-                .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
+            let _ = T::Court::on_dispute(
+                RawOrigin::Signed(caller.clone()).into(),
+                marketid,
+                OutcomeReport::Categorical(i.saturated_into())
+            );
         }
-    }: { Pallet::<T>::internal_resolve(&marketid)? }
```

Comment:
`internal_resolve` is a private function of `Court` and can't be called here. On the other hand, benchmark transposition to the `Court` pallet can't be applied because of the lack of some functions like https://github.com/zeitgeistpm/zeitgeist/blob/37aaf94def04d1323362dbea3c193cf09b093901/zrml/prediction-markets/src/benchmarks.rs#L78.

Not sure how to proceed here. Can you provide some thoughts @sea212 ?

Reply:
You can benchmark the `internal_resolve` variants in the court pallet. Then you can include the resulting weights module in the court pallet `pub mod weights`. After that you can include the weights module from the court pallet in the prediction market pallet and use it to calculate the actual weight in the relevant functions (`admin_move_market_to_closed`, `admin_move_market_to_resolved`):

```rust
// zrml/prediction-markets/src/lib.rs
// rename might be required
use zrml_court::weights::{WeightInfo, WeightInfoZeitgeist};

// -- snip --

let court_internal_resolve_weight = WeightInfo::<T>::internal_resolve_categorical_disputed(a, b, c, d);
// do weight adjustments
```

You also have to think about restructuring the benchmark functions based on the changes you made and about adding new benchmarks. For example, `on_initialize()` now became very complex and new benchmarks are required for a proper weight adjustment:

```rust
fn on_initialize(now: T::BlockNumber) -> Weight {
    let _ = T::Court::on_resolution(now);
    0
}
```

it calls the court pallet which in turn calls the (externally) private `internal_resolve` which is very complex. Instead of weight 0, it should return the proper benchmarked weight, which does depend on the weight calculations of the court pallet.